data race in testutil

```
WARNING: DATA RACE
Write by goroutine 50:
  github.com/coreos/etcd/pkg/testutil.(*RecorderBuffered).Record()
      /home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/pkg/testutil/recorder.go:49 +0x176
  github.com/coreos/etcd/store.(*storeRecorder).Get()
      /home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/store/store.go:774 +0x21a
  github.com/coreos/etcd/store.(*StoreRecorder).Get()
      <autogenerated>:38 +0xaf
  github.com/coreos/etcd/etcdserver.(*EtcdServer).applyRequest()
      github.com/coreos/etcd/etcdserver/_test/_obj_test/server.go:1276 +0x1528
  github.com/coreos/etcd/etcdserver.(*EtcdServer).apply()
      github.com/coreos/etcd/etcdserver/_test/_obj_test/server.go:1172 +0x533
  github.com/coreos/etcd/etcdserver.(*EtcdServer).applyEntries()
      github.com/coreos/etcd/etcdserver/_test/_obj_test/server.go:765 +0x520
  github.com/coreos/etcd/etcdserver.(*EtcdServer).applyAll()
      github.com/coreos/etcd/etcdserver/_test/_obj_test/server.go:663 +0x8a
  github.com/coreos/etcd/etcdserver.func·004()
      github.com/coreos/etcd/etcdserver/_test/_obj_test/server.go:545 +0xe4
Previous read by goroutine 89:
  github.com/coreos/etcd/pkg/testutil.(*RecorderBuffered).Wait()
      /home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/pkg/testutil/recorder.go:64 +0xa3
  github.com/coreos/etcd/store.(*storeRecorder).Wait()
      <autogenerated>:29 +0x9a
  github.com/coreos/etcd/etcdserver.TestApplySnapshotAndCommittedEntries()
      /home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server_test.go:997 +0x9b8
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:447 +0x133
Goroutine 50 (running) created at:
  github.com/coreos/etcd/etcdserver.func·005()
      github.com/coreos/etcd/etcdserver/_test/_obj_test/server.go:568 +0x5b4
Goroutine 89 (running) created at:
  testing.RunTests()
      /usr/local/go/src/testing/testing.go:555 +0xd4e
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:485 +0xe0
  main.main()
      github.com/coreos/etcd/etcdserver/_test/_testmain.go:242 +0x381
==================
--- FAIL: TestApplySnapshotAndCommittedEntries-4 (0.01s)
	server_test.go:1001: len(action) = 1, want 2
```

/cc @heyitsanthony

--- a/pkg/testutil/recorder.go
+++ b/pkg/testutil/recorder.go
@@ -61,7 +61,7 @@ func (r *RecorderBuffered) Wait(n int) (acts []Action, err error) {
 	WaitSchedule()
 	acts = r.Action()
 	if len(acts) < n {
-		err = newLenErr(n, len(r.actions))
+		err = newLenErr(n, len(acts))
 	}
 	return acts, err
 }

